int32 StartPoint
int32 EndPoint